src/dune_lang/blang.mli

```diff
@@ -13,19 +13,16 @@ module Op : sig
   val eval : t -> Ordering.t -> bool
 end
 
-type t =
+type 'string t =
```

Comment:
Do we instantiate this type with any other parameter?

Reply:
In the definition of Slang.form.

Reply:
blangs used in the slang language are `Slang.t Blang.t`. This is needed to handle the opam filters with nested conditional statements like:
```
build: [ "foo" { "%{var?bar:baz}%" = "bar" } ]
```
...where `var` is treated as a condition to choose between the strings `"bar"` and `"baz"` as part of a more complex condition `"%{var?bar:baz}%" = "bar"`.

Reply:
I see. If I understand this correctly, this expression corresponds to

```
(= (if (defined var) bar baz) bar)
```

Reply:
In any case, I do you think you should either

* Introduce your more general type as `Blang.Ast.t`
* Keep what you have, but introduce `Blang.Unexpanded.t` to contain the old type.

Reply:
Ok makes sense. I prefer your first suggestion. Also for the record the condition in the example above would be `(= (if (var bar baz) bar)`. Confusingly opam has two different uses for `?`. When it's used as a unary operator in a filter it tests if an expression is undefined but when it's used in a variable interpolation it separates a condition from values to use when the condition is true/false.